test/integration/change-streams/change_stream.test.ts

```diff
@@ -1434,240 +986,180 @@ describe('Change Streams', function () {
+      await changeStream.close();
+      await client?.close();
     });
 
     it('should work with events', {
       metadata: { requires: { topology: 'replicaset', mongodb: '>=4.1.1' } },
-      test: function (done) {
-        const changeStream = coll.watch([], { startAfter });
-        this.defer(() => changeStream.close());
-
-        coll.insertOne({ x: 2 }, { writeConcern: { w: 'majority', j: true } }, err => {
-          expect(err).to.not.exist;
-          changeStream.once('change', change => {
-            expect(change).to.containSubset({
-              operationType: 'insert',
-              fullDocument: { x: 2 }
-            });
+      async test() {
+        const willBeChange = once(changeStream, 'change');
+        await once(changeStream.cursor, 'init');
+        await collection.insertOne({ x: 2 });
 
-            done();
-          });
-        });
+        const [change] = await willBeChange;
+        expect(change).to.have.property('operationType', 'insert');
+        expect(change).to.have.nested.property('fullDocument.x', 2);
       }
     });
 
     it('should work with callbacks', {
       metadata: { requires: { topology: 'replicaset', mongodb: '>=4.1.1' } },
-      test: function (done) {
-        const changeStream = coll.watch([], { startAfter });
-        this.defer(() => changeStream.close());
+      async test() {
+        await initIteratorMode(changeStream);
 
-        coll.insertOne({ x: 2 }, { writeConcern: { w: 'majority', j: true } }, err => {
-          expect(err).to.not.exist;
-          exhaust(changeStream, [], (err, bag) => {
-            expect(err).to.not.exist;
-            const finalOperation = bag.pop();
-            expect(finalOperation).to.containSubset({
-              operationType: 'insert',
-              fullDocument: { x: 2 }
-            });
+        await collection.insertOne({ x: 2 });
 
-            done();
-          });
-        });
+        const change = await changeStream.next();
+        expect(change).to.have.property('operationType', 'insert');
+        expect(change).to.have.nested.property('fullDocument.x', 2);
       }
     });
   });
 
   describe('Change Stream Resume Error Tests', function () {
     it('should continue emitting change events after a resumable error', {
-      metadata: { requires: { topology: 'replicaset', mongodb: '>=3.6' } },
-      test: withChangeStream((collection, changeStream, done) => {
-        const docs = [];
-        changeStream.on('change', change => {
-          expect(change).to.exist;
-          docs.push(change);
-          if (docs.length === 2) {
-            expect(docs[0]).to.containSubset({
-              operationType: 'insert',
-              fullDocument: { a: 42 }
-            });
-            expect(docs[1]).to.containSubset({
-              operationType: 'insert',
-              fullDocument: { b: 24 }
-            });
-            done();
+      metadata: { requires: { topology: 'replicaset', mongodb: '>3.6' } },
```

Comment:
```suggestion
      metadata: { requires: { topology: 'replicaset' } },
```

Reply:
`>3.6` holds significance here and the other cases, this test doesn't pass on 3.6 it did not seem worth digging into why for now. I can file TODOs but the tests themselves haven't been evaluated for merit so I feel like it's an acceptable gap. Thoughts?

Reply:
Agreed, I think that's fine.  Thanks for explaining

Reply:
continuing to emit change events after a resumable error seems important; 

and I disagree with the general approach of changing the requirements to `>= 3.6` elsewhere - either the test is worth having, in which case it needs to work on all our supported servers, including 3.6, or it's not worth having, in which case we should be getting rid of it entirely; doing `>3.6` just makes it confusing and immediately raises the question of why the test isn't applicable on 3.6.

there are a total of 6 tests that have been updated in this way - it should not add that much scope to read through them and figure out whether the cases as stated are worth testing; and for the stated cases that are worth testing, to determine whether the approach in the test is correct or should be adjusted, and then either make the changes or file follow ups as needed

Reply:
3.6 doesn't emit invalidate events which some of these tests rely on, so I've marked those as 4.2+
The resumability ones timeout on 3.6, investigating, stay tuned

Reply:
but why did the original tests work? or were they skipped? or just not asserting correctly?

Reply:
The resume tests are strange, they don't simulate the network error coming from the network calls, I had Bailey look over my shoulder for this one and he's pointing out that the tests `emit('error'` which circumvents all the cursor cleanup logic that would be done by abstract cursor. So I've skipped them with a todo of the fixing resumability ticket we're getting to soon.

And for the invalidate one's sorry I flipped it, rename is the event that was introduced in 4.0.1, the test before didn't look for rename it just ran the event listener until it saw invalidate and then would call done, so I have added a new assertion since with promises I have to iterate passed the rename event, I can revert this but I think we on the border of testing server behavior here anyway.